core/src/main/java/io/grpc/transport/inprocess/InProcessTransport.java

```diff
@@ -137,18 +138,18 @@ public synchronized void shutdown() {
       return;
     }
     shutdownStatus = Status.UNAVAILABLE.withDescription("transport was requested to shut down");
-    notifyShutdown();
+    notifyShutdown(shutdownStatus);
```

Comment:
We could add some special Statuses that are constants and use reference equality to check if they are being passed if need be.  UNAVAILABLE typically means something that is safe to retry.